drv/stm32h7-fmc-demo-server/src/main.rs

```diff
@@ -226,15 +251,57 @@ fn main() -> ! {
 
 struct ServerImpl {
     fmc: &'static device::fmc::RegisterBlock,
+    net: Net,
+    rx_packet: &'static mut [u8; 1024],
+    tx_packet: &'static mut [u8; 1024],
 }
 
 impl NotificationHandler for ServerImpl {
     fn current_notification_mask(&self) -> u32 {
-        0
+        notifications::SOCKET_MASK
     }
 
-    fn handle_notification(&mut self, _bits: u32) {
-        unreachable!()
+    fn handle_notification(&mut self, bits: u32) {
+        if bits & notifications::SOCKET_MASK != 0 {
+            const SOCKET: SocketName = SocketName::fmc_test;
+            loop {
+                match self.net.recv_packet(
+                    SOCKET,
+                    LargePayloadBehavior::Discard,
+                    self.rx_packet,
+                ) {
+                    Ok(mut meta) => {
+                        count!(Event::RecvPacket);
+                        let data = &mut self.rx_packet[..meta.size as usize];
+                        match process_network_packet(data, self.tx_packet) {
+                            Ok(size) => {
+                                meta.size = size as u32;
+                                count!(Event::Respond);
+                            }
+                            Err(e) => {
+                                count!(Event::RequestRejected);
+                                self.tx_packet[0] = e as u8;
+                                meta.size = 1;
+                            }
+                        }
+                        let response = &self.tx_packet[..meta.size as usize];
+                        // We're going to send best-effort because the only
+                        // thing that can really go wrong here is filling our
+                        // own queue, which, well, send slower.
+                        let _ = self.net.send_packet(SOCKET, meta, response);
```

Comment:
This seems like it could be confusing for users, who will see packets being dropped if they send too fast.  Is there a downside to looping here?

Reply:
FWIW, this isn't what will cause users to see packets getting dropped. Because the incoming and outgoing queues are the same size, sending too fast is far more likely to drop on receipt than response.

That being said, I can add the loop here. I was being lazy and felt the indentation was getting a little deep. :-)

Reply:
So this has caused me to realize that there's a small subtle liveness bug in our normal retry loop pattern that is unlikely to occur in practice, but is theoretically possible:
- A task attempts to send
- It finds its queue full
- It begins waiting for the net stack's obligatory "your queue is no longer full" notification
- The net stack _crashes_ and is restarted
- The net stack observes all queues as having space available and goes on processing packets.

In this case, the transmitting task will sit and wait until it receives a net notification, which in this case is likely to be the next packet addressed to one of its sockets.

I have recorded this as #1844.